examples/mqtt-dashboard/device/main.c

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2023 Cesanta Software Limited
+// All rights reserved
+//
+// Example MQTT client. It performs the following steps:
+//
+// To enable SSL/TLS, see https://mongoose.ws/tutorials/tls/#how-to-build
+
+#include "mongoose.h"
+
+#define DEVICE_ID_LEN 10
```

Comment:
I suggest to get rid of this. Instead of fixed-size buffers, use malloced strings.

```c
#define MQTT_SERVER_URL "...."
static const char *s_url;

int main() {
  s_url = strdup(MQTT_SERVER_URL);  // Set default

  for (i = 1; i < argc; i++) {
    if (strcmp(argv[i], "-u") == 0 && argv[i + 1] != NULL) {
      free(s_url);
      s_url = strdup(argv[++i]);
  } ...
}
```